pkg/ebpf/gadgets/http/tracer/helpers.go

```diff
@@ -0,0 +1,183 @@
+package tracer
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"net"
+	"net/http"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+	tracerhttptype "github.com/kubescape/node-agent/pkg/ebpf/gadgets/http/types"
+)
+
+func parseHTTPRequest(data []byte) (tracerhttptype.HTTPRequestData, error) {
+	bufReader := bufio.NewReader(bytes.NewReader(data))
+
+	req, err := http.ReadRequest(bufReader)
+	if err != nil {
+		return tracerhttptype.HTTPRequestData{}, err
+	}
+	defer req.Body.Close()
+	headers := req.Header.Clone()
+	headers.Set("Host", req.Host)
+
+	return tracerhttptype.HTTPRequestData{
+		Method:  req.Method,
+		URL:     req.URL.String(),
+		Headers: headers,
+	}, nil
+}
+
+func parseHTTPResponse(data []byte) (tracerhttptype.HTTPResponseData, error) {
+	bufReader := bufio.NewReader(bytes.NewReader(data))
+
+	statusLine, err := bufReader.ReadString('\n')
+	if err != nil {
+		return tracerhttptype.HTTPResponseData{}, fmt.Errorf("error reading status line: %v", err)
+	}
+
+	parts := strings.SplitN(strings.TrimSpace(statusLine), " ", 3)
+	if len(parts) < 3 {
+		return tracerhttptype.HTTPResponseData{}, fmt.Errorf("invalid status line: %s", statusLine)
+	}
+
+	statusCode, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return tracerhttptype.HTTPResponseData{}, fmt.Errorf("invalid status code: %v", err)
+	}
+
+	headers := make(http.Header)
+	for {
+		line, err := bufReader.ReadString('\n')
+		if err != nil {
+			break
+		}
+
+		line = strings.TrimSpace(line)
+		if line == "" {
+			break
+		}
+
+		parts := strings.SplitN(line, ":", 2)
+		if len(parts) != 2 {
+			continue
+		}
+
+		key := strings.TrimSpace(parts[0])
+		value := strings.TrimSpace(parts[1])
+		headers.Add(key, value)
+	}
+
+	return tracerhttptype.HTTPResponseData{
+		StatusCode: statusCode,
+		Status:     strings.Join(parts[2:], " "),
+		Headers:    headers,
+	}, nil
+}
+
+func FromCString(in []byte) []byte {
```

Comment:
I think IG have their own function for this that we use in other gadgets as well.

Reply:
They do have it for strings only